# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxPlatformFontList.cpp
# Commit: c03c7df7cd61
# Full Hash: c03c7df7cd61c88de130f49dbceaa8a62c9a7758
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-05-14 21:58:21
# Regressor Bug: 1708768
# File Overlap Count: 2
# Description:
#   Bug 1708768 - Preliminary cleanup of gfxPlatformFontList creation/initialization code; no functional change. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D114863
# ==============================================================================

diff -r 76dd6379cc01 -r c03c7df7cd61 gfx/thebes/gfxPlatformFontList.cpp
--- a/gfx/thebes/gfxPlatformFontList.cpp	Fri May 14 17:45:16 2021 +0000
+++ b/gfx/thebes/gfxPlatformFontList.cpp	Fri May 14 18:05:44 2021 +0000
@@ -436,7 +436,7 @@
   return added;
 }
 
-nsresult gfxPlatformFontList::InitFontList() {
+bool gfxPlatformFontList::InitFontList() {
   // This shouldn't be called from stylo threads!
   MOZ_ASSERT(NS_IsMainThread());
 
@@ -486,8 +486,6 @@
   mVisibilityLevel = FontVisibility::Unknown;
   SetVisibilityLevel();
 
-  sPlatformFontList = this;
-
   // Try to initialize the cross-process shared font list if enabled by prefs,
   // but not if we're running in Safe Mode.
   if (StaticPrefs::gfx_e10s_font_list_shared_AtStartup() &&
@@ -516,18 +514,15 @@
                          "falling back to in-process list.";
       mSharedFontList.reset(nullptr);
     }
-    if (oldSharedList) {
-      if (XRE_IsParentProcess()) {
-        // notify all children of the change
-        dom::ContentParent::NotifyUpdatedFonts(true);
-      }
+    if (oldSharedList && XRE_IsParentProcess()) {
+      // notify all children of the change
+      dom::ContentParent::NotifyUpdatedFonts(true);
     }
   }
 
   if (!SharedFontList()) {
-    nsresult rv = InitFontListForPlatform();
-    if (NS_FAILED(rv)) {
-      return rv;
+    if (NS_FAILED(InitFontListForPlatform())) {
+      return false;
     }
     ApplyWhitelist();
   }
@@ -538,16 +533,13 @@
   FontFamily fam = GetDefaultFont(&defStyle);
   if (fam.mIsShared) {
     auto face = fam.mShared->FindFaceForStyle(SharedFontList(), defStyle);
-    if (!face) {
-      mDefaultFontEntry = nullptr;
-    } else {
-      mDefaultFontEntry = GetOrCreateFontEntry(face, fam.mShared);
-    }
+    mDefaultFontEntry =
+        face ? GetOrCreateFontEntry(face, fam.mShared) : nullptr;
   } else {
     mDefaultFontEntry = fam.mUnshared->FindFontForStyle(defStyle);
   }
 
-  return NS_OK;
+  return true;
 }
 
 void gfxPlatformFontList::InitializeCodepointsWithNoFonts() {